website/static/js/registrationUtils.js

```diff
@@ -779,7 +781,54 @@ RegistrationEditor.prototype.save = function() {
     }
     return true;
 };
+/**
+ * Returns the `user_fullname` of each contributor attached to a node.
+ **/
+RegistrationEditor.prototype.getContributors = function() {
+    var contributorsUrl = window.contextVars.node.urls.api + 'contributors_abbrev/';
+    var contributorNames = [];
+    $osf.ajaxJSON('get', contributorsUrl).done(function(data) {
+        $.each(data.contributors, function(idx, val) {
+            contributorNames.push(val.user_fullname);
+        });
+    });
+    return contributorNames;
+};
+/**
+ * Opens a bootbox dialog with a checkbox list of each contributor
+ * the user has the option to import all contributors or to select
+ * each one individually.
+ **/
+RegistrationEditor.prototype.authorDialog = function() {
+    var self = this;
 
+    bootbox.dialog({
+        title: 'Choose which contributors to import:',
+        message: function() {
+            ko.renderTemplate('importContributors', self, {}, this, 'replaceNode');
+        },
+        buttons: {
+            select: {
+                label: 'Import',
+                className: 'btn-primary pull-left',
+                callback: function() {
+                    var boxes = document.querySelectorAll('input[type="checkbox"]');
```

Comment:
Honest question here: will this also pick up checkboxes that are part of the page outside the modal? If so you may want to limit your selector to a specific container.